File: VB/SilverlightApplication15/PageViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Windows.Input
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.Core.Commands
Imports DevExpress.Xpf.Printing
' ...

Partial Public Class PageViewModel
    Implements INotifyPropertyChanged

    Public Property ReportPreviewModel() As ReportPreviewModel

    Public Property SubreportParameter() As String
        Get
            Dim param = ReportPreviewModel.Parameters("xrSubreport1.parameter1")
            Return If(param IsNot Nothing, CStr(param.Value), String.Empty)
        End Get
        Set(ByVal value As String)
            ReportPreviewModel.Parameters("xrSubreport1.parameter1").Value = value
            RaisePropertyChanged("SubreportParameter")
        End Set
    End Property

    Public Property RecreateDocument() As ICommand

    Public Sub New()
        RecreateDocument = New DelegateCommand(Of Object)(AddressOf DoRecreateDocument)
        ReportPreviewModel = New ReportPreviewModel("../ReportService1.svc") With {.ReportName = "SilverlightApplication15.Web.MasterReport, SilverlightApplication15.Web"}
    End Sub

    Private Sub DoRecreateDocument(ByVal parameter As Object)
        ReportPreviewModel.CreateDocument()
    End Sub
End Class

File: CS/SilverlightApplication15/PageViewModel.cs
using System;
using System.ComponentModel;
using System.Linq.Expressions;
using System.Windows.Input;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.Core.Commands;
using DevExpress.Xpf.Printing;
// ...

namespace SilverlightApplication15 {
    public partial class PageViewModel : INotifyPropertyChanged {
        public ReportPreviewModel ReportPreviewModel { get; private set; }
        public ICommand RecreateDocument { get; private set; }

        public string SubreportParameter {
            get {
                var param = ReportPreviewModel.Parameters["xrSubreport1.parameter1"];
                return param != null ? (string)param.Value : string.Empty;
            }
            set {
                ReportPreviewModel.Parameters["xrSubreport1.parameter1"].Value = value;
                RaisePropertyChanged("SubreportParameter");
            }
        }

        public PageViewModel() {
            RecreateDocument = new DelegateCommand<object>(DoRecreateDocument);
            ReportPreviewModel = new ReportPreviewModel("../ReportService1.svc") {
                ReportName = "SilverlightApplication15.Web.MasterReport, SilverlightApplication15.Web"
            };
        }

        void DoRecreateDocument(object o) {
            ReportPreviewModel.CreateDocument();
        }
    }
}
